### ISSUE TITLE
Cryptic error when leaving @inheritDotParams unspecified.

### ISSUE THREAD

**Issue by Ax3man** [r-lib/roxygen2#670]
By mistake I left in a `@inheritDotParams` line, with no function specified from which to inherit. I did not notice for some time, since `devtools::load_all(".")` was working fine.

However, when building or checking the package it suddenly stopped early with the following error:

```
Updating phylopath documentation
Loading phylopath
Error: is.function(fun) is not TRUE
Execution halted

Exited with status 1.
```

It was not clear to me what was wrong. I traced the error as follows:

- I figured out that `devtools::document()` also wasn't working.
- `roxygen2::roxygenise` was failing.
- I used the debugger to step through:  `roxygenise`, `roc_out`, `roclet_process` and `topics_process_inherit` to find that `topics$apply(inherit_dot_params, env = env)` was failing.

Would it be possible to report the topic for which the error occured, or at least that some inheritance failed? That would make this much easier to pin down.

Thanks all for a massive time saver.

**Comment by hadley** [comment]
Reproducible example:

```R
  out <- roc_proc_text(rd_roclet(), "
    #' Foo
    #' @inheritDotParams  
    foo <- function(...) {}
  ")[[1]]
```

Note that this requires two spaces after `@inheritDotParams` to trigger

**Comment by hadley** [comment]
Generally, looks like all the inherits tags have poorly defined failure modes. 

Should move the more dangerous code into the tag parsing, returning a richer object, i.e. move logic our of `inherit_dot_params` and `select_args_text` into a custom tag parser.

**Comment by hadley** [comment]
That's not 100% straightforward because the inherits tags need the code environment, which does not exist during code parsing; and by the time we get to `select_args_text` we no longer have information about the location of the tag.

**Comment by hadley** [comment]
``` r
library(roxygen2)
out <- roc_proc_text(rd_roclet(), "
  #' Foo
  #' @inheritDotParams  
  foo <- function(...) {}
")[[1]]
#> Warning: @inheritDotParams [/tmp/RtmpFOir1O/filefe603a93250d#4]: requires a
#> value
```

<sup>Created on 2019-08-12 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Looks like this was fixed as part of #869
